nav2_util/include/nav2_util/geometry_utils.hpp

```diff
@@ -107,6 +124,25 @@ inline Iter min_by(Iter begin, Iter end, Getter getCompareVal)
   return lowest_it;
 }
 
+/**
+ * Find first element in iterator that is greater integrated distance than comparevalue
+ */
+template<typename Iter, typename Getter>
+inline Iter first_element_beyond(Iter begin, Iter end, Getter getCompareVal)
```

Comment:
I still wonder if there is a more descriptive name for this function, something that clarifies that we are using integrated distance. Maybe `find_after_integrated_distance`?

Reply:
How about `first_element_after_integrated_distance` or `first_after_integrated_distance`? That's alot of words, but the most clear combining the two. I think the "first" word (or one like it) is important to communicate its returning the first single item that meets the criteria, not any arbitrary one or a set of them. `find_` I think could be misinterpreted as potentially returning all.

Reply:
I like `*_after_integrated_distance`. I'm not as concerned about `find` in this context, since in most programming languages (including C++), `find` means finding the first match. But I think `first_after_integrated_distance` is just fine.

Reply:
OK, lets do that